### ISSUE TITLE
Determine final rules for Dates object requirements

### ISSUE THREAD

**Issue by p-galligan** [RockefellerArchiveCenter/rac-data-model#35]
We want to require begin dates when possible, but ArchivesSpace doesn't require begin dates on date objects. 

Should we allow "null" dates for begin dates? Or should we parse date expressions to begin dates?

**Comment by helrond** [comment]
I think there are three options:
- Map whatever date data is available in ArchivesSpace directly to the RAC data model. In this case, we'd have to loosen up our schema to match the AS date data model, which would mean allowing non-structured dates.
- Do some parsing of the date expression to extract structured dates where possible. Although we'll get better structured date coverage this way, we won't get coverage for "undated" or un-parseable dates, so ultimately we'll still have to loosen up the schema.
- Parse the date expression AND inherit dates from the closest date parseable parent where necessary. This is (I think) a lot more work, and there may be performance considerations, but would result in us having much better structured data info.

I know @bonniegee has gotten into this with the digitization dates, so tagging her in in case she has advice/relevant code snippets.

**Comment by bonniegee** [comment]
I have parsed date expressions to get end years, but it is not very elegant and allows for a fair amount of error. (Which was fine in the case I was using it for, but would need to be refined for this used.) When a structured date doesn't exist AND the date expression was not "undated," I parsed the date expression from the end until I got a string of four digits. [Relevant code is here](https://github.com/RockefellerArchiveCenter/scripts/blob/master/archivessnake/findByRefid.py#L28). I also ran the find_year() function over titles and certain notes (like scope and content) because that was uncovering information, but that would be more relevant to data cleanup than to this.
